datafusion/physical-plan/src/sorts/merge.rs

```diff
@@ -266,6 +319,35 @@ impl<C: CursorValues> SortPreservingMergeStream<C> {
         }
     }
 
+    #[inline]
+    fn is_eq_and_gt(&self, a: usize, b: usize) -> (bool, bool) {
```

Comment:
Can match on or return [`cmp` ](https://doc.rust-lang.org/std/cmp/enum.Ordering.html) instead? This will save one comparison :)

Reply:
Ah this compares the partition index rather than value? Why is that needed, might want to add some documentation here?

Reply:
cursor[a] is the value, so we compare value actually

> Can match on and/or return [Ordering ](https://doc.rust-lang.org/std/cmp/enum.Ordering.html)instead? This will save one comparison :)

Not sure what do you mean 🤔